diff --git a/node_modules/@formatjs/intl/src/types.d.ts b/node_modules/@formatjs/intl/src/types.d.ts
index 1f73905..9abaacd 100644
--- a/node_modules/@formatjs/intl/src/types.d.ts
+++ b/node_modules/@formatjs/intl/src/types.d.ts
@@ -8,10 +8,8 @@ import { DEFAULT_INTL_CONFIG } from './utils';
 import { NumberFormatOptions } from '@formatjs/ecma402-abstract';
 declare global {
     namespace FormatjsIntl {
-        interface Message {
-        }
-        interface IntlConfig {
-        }
+        interface Message { ids?: string }
+        interface IntlConfig { locale?: string }
         interface Formats {
         }
     }
