diff --git a/node_modules/@storybook/router/dist/index.d.ts b/node_modules/@storybook/router/dist/index.d.ts
index ed699f7..71f48eb 100644
--- a/node_modules/@storybook/router/dist/index.d.ts
+++ b/node_modules/@storybook/router/dist/index.d.ts
@@ -335,7 +335,7 @@ declare const useNavigate: () => (to: To | number, { plain, ...options }?: any)
  *  A component that will navigate to a new location/path when clicked
  */
 declare const Link: {
-    ({ to, children, ...rest }: LinkProps): React__default.JSX.Element;
+    ({ to, children, ...rest }: LinkProps): JSX.Element;
     displayName: string;
 };
 /**
@@ -343,7 +343,7 @@ declare const Link: {
  * and will be called whenever it changes when it changes
  */
 declare const Location: {
-    ({ children }: LocationProps): React__default.JSX.Element;
+    ({ children }: LocationProps): JSX.Element;
     displayName: string;
 };
 /**
