diff --git a/node_modules/@storybook/manager-api/dist/index.d.ts b/node_modules/@storybook/manager-api/dist/index.d.ts
index 7afb86e..a15b801 100644
--- a/node_modules/@storybook/manager-api/dist/index.d.ts
+++ b/node_modules/@storybook/manager-api/dist/index.d.ts
@@ -822,7 +822,7 @@ declare class ManagerProvider extends Component<ManagerProviderProps, State> {
     static getDerivedStateFromProps(props: ManagerProviderProps, state: State): State;
     shouldComponentUpdate(nextProps: ManagerProviderProps, nextState: State): boolean;
     initModules: () => void;
-    render(): React.JSX.Element;
+    render(): JSX.Element;
 }
 interface ManagerConsumerProps<P = unknown> {
     filter?: (combo: Combo) => P;
