diff --git a/node_modules/@types/express/index.d.ts b/node_modules/@types/express/index.d.ts
index 8427f02..2b4698f 100755
--- a/node_modules/@types/express/index.d.ts
+++ b/node_modules/@types/express/index.d.ts
@@ -55,7 +55,7 @@ declare namespace e {
     /**
      * This is a built-in middleware function in Express. It serves static files and is based on serve-static.
      */
-    var static: serveStatic.RequestHandlerConstructor<Response>;
+    var static: unknown;
 
     /**
      * This is a built-in middleware function in Express. It parses incoming requests with urlencoded payloads and is based on body-parser.
