diff --git a/index.d.ts b/index.d.ts
index f95f9c1747a4cc6229b29ab450a859277ad3c16e..6650687fe938868555bb7ec2dbc67ea4a219a0e9 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -3,7 +3,8 @@
 import FormData = require("form-data");
 import { RequestOptions } from "http";
 import { URL, URLSearchParams } from "url";
-import { AbortSignal } from "./externals";
+// import { AbortSignal } from "./externals";
+// Avoid type conflicts!
 
 declare class Request extends Body {
     constructor(input: RequestInfo, init?: RequestInit);
@@ -130,7 +131,7 @@ declare class Body {
     blob(): Promise<Blob>;
     body: NodeJS.ReadableStream;
     bodyUsed: boolean;
-    buffer(): Promise<Buffer>;
+    buffer(): Promise<Buffer<ArrayBuffer>>;
     json(): Promise<any>;
     size: number;
     text(): Promise<string>;
