diff --git a/node_modules/@types/blueimp-load-image/index.d.ts b/node_modules/@types/blueimp-load-image/index.d.ts
index 285505b..da92b91 100644
--- a/node_modules/@types/blueimp-load-image/index.d.ts
+++ b/node_modules/@types/blueimp-load-image/index.d.ts
@@ -9,7 +9,7 @@
 declare namespace loadImage {
     type LoadImageCallback = (eventOrImage: Event | HTMLCanvasElement | HTMLImageElement, data?: MetaData) => void;
     type LoadImageResult = MetaData & {
-        image: HTMLImageElement | FileReader | false;
+        image: HTMLImageElement | HTMLCanvasElement;
     };

     type ParseMetaDataCallback = (data: MetaData) => void;
@@ -122,6 +122,11 @@ interface LoadImage {
     ) => void;

     blobSlice: (this: Blob, start?: number, end?: number) => Blob;
+
+  scale: (
+      img: HTMLImageElement | HTMLCanvasElement,
+      options?: loadImage.LoadImageOptions
+  ) => HTMLImageElement | HTMLCanvasElement;
 }

 declare const loadImage: LoadImage;
