diff --git a/node_modules/@types/node-fetch/index.d.ts b/node_modules/@types/node-fetch/index.d.ts
index 346d0b2..8b3efdd 100755
--- a/node_modules/@types/node-fetch/index.d.ts
+++ b/node_modules/@types/node-fetch/index.d.ts
@@ -19,7 +19,8 @@
 import FormData = require('form-data');
 import { RequestOptions } from "http";
 import { URLSearchParams, URL } from "url";
-import { AbortSignal } from "./externals";
+// Avoid type conflicts!
+// import { AbortSignal } from "./externals";
 
 export class Request extends Body {
     constructor(input: RequestInfo, init?: RequestInit);
