diff --git a/node_modules/@types/jest/index.d.ts b/node_modules/@types/jest/index.d.ts
index 272822a..87ebd1f 100755
--- a/node_modules/@types/jest/index.d.ts
+++ b/node_modules/@types/jest/index.d.ts
@@ -30,18 +30,18 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // Minimum TypeScript Version: 3.8

-declare var beforeAll: jest.Lifecycle;
-declare var beforeEach: jest.Lifecycle;
-declare var afterAll: jest.Lifecycle;
-declare var afterEach: jest.Lifecycle;
-declare var describe: jest.Describe;
-declare var fdescribe: jest.Describe;
-declare var xdescribe: jest.Describe;
-declare var it: jest.It;
-declare var fit: jest.It;
-declare var xit: jest.It;
-declare var test: jest.It;
-declare var xtest: jest.It;
+// declare var beforeAll: jest.Lifecycle;
+// declare var beforeEach: jest.Lifecycle;
+// declare var afterAll: jest.Lifecycle;
+// declare var afterEach: jest.Lifecycle;
+// declare var describe: jest.Describe;
+// declare var fdescribe: jest.Describe;
+// declare var xdescribe: jest.Describe;
+// declare var it: jest.It;
+// declare var fit: jest.It;
+// declare var xit: jest.It;
+// declare var test: jest.It;
+// declare var xtest: jest.It;

 declare const expect: jest.Expect;

