diff --git a/node_modules/@types/node/tls.d.ts b/node_modules/@types/node/tls.d.ts
index 2c55eb9..a594969 100755
--- a/node_modules/@types/node/tls.d.ts
+++ b/node_modules/@types/node/tls.d.ts
@@ -621,6 +621,12 @@ declare module 'tls' {
          * `identity` must use UTF-8 encoding.
          */
         pskCallback?(hint: string | null): PSKCallbackNegotation | null;
+
+        /* Node.js documentation says:
+         * "...: Any socket.connect() option not already listed."
+         * and "signal" is one of them.
+         */
+        signal?: AbortSignal;
     }
     /**
      * Accepts encrypted connections using TLS or SSL.
