diff --git a/index.d.ts b/index.d.ts
index 4fed6b906ef0d56b429ccaa9e2804915175e6f7e..4097e01c2d6ca029248daea20ae82c85b35a2a46 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -487,7 +487,7 @@ declare global {
              * @param expected   Potential expected value.
              * @param message   Message to display on error.
              */
-            strictEqual<T>(actual: T, expected: T, message?: string): void;
+            strictEqual<T>(actual: unknown, expected: T, message?: string): asserts actual is T;
 
             /**
              * Asserts strict inequality (!==) of actual and expected.
