diff --git a/node_modules/app-builder-lib/out/targets/ArchiveTarget.js b/node_modules/app-builder-lib/out/targets/ArchiveTarget.js
index 72a59df..ba57ef5 100644
--- a/node_modules/app-builder-lib/out/targets/ArchiveTarget.js
+++ b/node_modules/app-builder-lib/out/targets/ArchiveTarget.js
@@ -118,10 +118,6 @@ class ArchiveTarget extends _core().Target {
         withoutDir
       };
       await (0, _archive().archive)(format, artifactPath, dirToArchive, archiveOptions);
-
-      if (this.isWriteUpdateInfo && format === "zip") {
-        updateInfo = await (0, _differentialUpdateInfoBuilder().appendBlockmap)(artifactPath);
-      }
     }

     await packager.info.callArtifactBuildCompleted({
diff --git a/node_modules/app-builder-lib/out/targets/LinuxTargetHelper.js b/node_modules/app-builder-lib/out/targets/LinuxTargetHelper.js
index 6e674b9..8e5be4f 100644
--- a/node_modules/app-builder-lib/out/targets/LinuxTargetHelper.js
+++ b/node_modules/app-builder-lib/out/targets/LinuxTargetHelper.js
@@ -135,7 +135,7 @@ class LinuxTargetHelper {
         exec += executableArgs.join(" ");
       }

-      exec += " %U";
+      exec += " --no-sandbox %U";
     }

     const desktopMeta = Object.assign(Object.assign({
