diff --git a/out/dmgLicense.js b/out/dmgLicense.js
index 9263c2b..40706e5 100644
--- a/out/dmgLicense.js
+++ b/out/dmgLicense.js
@@ -6,7 +6,7 @@ const js_yaml_1 = require("js-yaml");
 const license_1 = require("app-builder-lib/out/util/license");
 const fs_extra_1 = require("fs-extra");
 const licenseButtons_1 = require("./licenseButtons");
-const dmg_license_1 = require("dmg-license");
+// SIGNAL: const dmg_license_1 = require("dmg-license");
 async function addLicenseToDmg(packager, dmgPath) {
     const licenseFiles = await (0, license_1.getLicenseFiles)(packager);
     if (licenseFiles.length === 0) {
