diff --git a/node_modules/react-blurhash/lib/Blurhash.js b/node_modules/react-blurhash/lib/Blurhash.js
index db3115d..746c877 100644
--- a/node_modules/react-blurhash/lib/Blurhash.js
+++ b/node_modules/react-blurhash/lib/Blurhash.js
@@ -61,8 +61,8 @@ var Blurhash = /** @class */ (function (_super) {
     };
     Blurhash.prototype.render = function () {
         var _a = this.props, hash = _a.hash, height = _a.height, width = _a.width, punch = _a.punch, resolutionX = _a.resolutionX, resolutionY = _a.resolutionY, style = _a.style, rest = __rest(_a, ["hash", "height", "width", "punch", "resolutionX", "resolutionY", "style"]);
-        return (react_1.default.createElement("div", __assign({}, rest, { style: __assign({ display: 'inline-block', height: height, width: width }, style, { position: 'relative' }) }),
-            react_1.default.createElement(BlurhashCanvas_1.default, { hash: hash, height: resolutionY, width: resolutionX, punch: punch, style: canvasStyle })));
+        return (react_1.createElement("div", __assign({}, rest, { style: __assign({ display: 'inline-block', height: height, width: width }, style, { position: 'relative' }) }),
+            react_1.createElement(BlurhashCanvas_1.default, { hash: hash, height: resolutionY, width: resolutionX, punch: punch, style: canvasStyle })));
     };
     Blurhash.defaultProps = {
         height: 128,
@@ -71,6 +71,6 @@ var Blurhash = /** @class */ (function (_super) {
         resolutionY: 32,
     };
     return Blurhash;
-}(react_1.default.PureComponent));
+}(react_1.PureComponent));
 exports.default = Blurhash;
 //# sourceMappingURL=Blurhash.js.map
\ No newline at end of file
diff --git a/node_modules/react-blurhash/lib/BlurhashCanvas.js b/node_modules/react-blurhash/lib/BlurhashCanvas.js
index b2833dc..d666520 100644
--- a/node_modules/react-blurhash/lib/BlurhashCanvas.js
+++ b/node_modules/react-blurhash/lib/BlurhashCanvas.js
@@ -63,13 +63,13 @@ var BlurhashCanvas = /** @class */ (function (_super) {
     };
     BlurhashCanvas.prototype.render = function () {
         var _a = this.props, hash = _a.hash, height = _a.height, width = _a.width, rest = __rest(_a, ["hash", "height", "width"]);
-        return react_1.default.createElement("canvas", __assign({}, rest, { height: height, width: width, ref: this.handleRef }));
+        return react_1.createElement("canvas", __assign({}, rest, { height: height, width: width, ref: this.handleRef }));
     };
     BlurhashCanvas.defaultProps = {
         height: 128,
         width: 128,
     };
     return BlurhashCanvas;
-}(react_1.default.PureComponent));
+}(react_1.PureComponent));
 exports.default = BlurhashCanvas;
 //# sourceMappingURL=BlurhashCanvas.js.map
\ No newline at end of file
