diff --git a/node_modules/react-aria-components/dist/types.d.ts b/node_modules/react-aria-components/dist/types.d.ts
index eb908b4..6cd530f 100644
--- a/node_modules/react-aria-components/dist/types.d.ts
+++ b/node_modules/react-aria-components/dist/types.d.ts
@@ -44,7 +44,7 @@ interface SlotProps {
     /** A slot name for the component. Slots allow the component to receive props from a parent component. */
     slot?: string;
 }
-export function useContextProps<T, U, E extends Element>(props: T & SlotProps, ref: React.ForwardedRef<E>, context: React.Context<ContextValue<U, E>>): [T, React.RefObject<E>];
+export function useContextProps<T, U extends SlotProps, E extends Element>(props: T & SlotProps, ref: React.ForwardedRef<E>, context: React.Context<ContextValue<U, E>>): [T, React.RefObject<E>];
 interface CollectionProps<T> extends Omit<CollectionBase<T>, 'children'> {
     /** The contents of the collection. */
     children?: ReactNode | ((item: T) => ReactElement);
