diff --git a/node_modules/react-textarea-autosize/dist/react-textarea-autosize.cjs.dev.js b/node_modules/react-textarea-autosize/dist/react-textarea-autosize.cjs.dev.js
index ce25001..36bcd17 100644
--- a/node_modules/react-textarea-autosize/dist/react-textarea-autosize.cjs.dev.js
+++ b/node_modules/react-textarea-autosize/dist/react-textarea-autosize.cjs.dev.js
@@ -110,7 +110,7 @@ var pick = function pick(props, obj) {
 var SIZING_STYLE = ['borderBottomWidth', 'borderLeftWidth', 'borderRightWidth', 'borderTopWidth', 'boxSizing', 'fontFamily', 'fontSize', 'fontStyle', 'fontWeight', 'letterSpacing', 'lineHeight', 'paddingBottom', 'paddingLeft', 'paddingRight', 'paddingTop', // non-standard
 'tabSize', 'textIndent', // non-standard
 'textRendering', 'textTransform', 'width', 'wordBreak'];
-var isIE = typeof document !== 'undefined' ? !!document.documentElement.currentStyle : false;
+var isIE = false;

 var getSizingData = function getSizingData(node) {
   var style = window.getComputedStyle(node);
diff --git a/node_modules/react-textarea-autosize/dist/react-textarea-autosize.cjs.prod.js b/node_modules/react-textarea-autosize/dist/react-textarea-autosize.cjs.prod.js
index d4e39a2..f26641e 100644
--- a/node_modules/react-textarea-autosize/dist/react-textarea-autosize.cjs.prod.js
+++ b/node_modules/react-textarea-autosize/dist/react-textarea-autosize.cjs.prod.js
@@ -110,7 +110,7 @@ var pick = function pick(props, obj) {
 var SIZING_STYLE = ['borderBottomWidth', 'borderLeftWidth', 'borderRightWidth', 'borderTopWidth', 'boxSizing', 'fontFamily', 'fontSize', 'fontStyle', 'fontWeight', 'letterSpacing', 'lineHeight', 'paddingBottom', 'paddingLeft', 'paddingRight', 'paddingTop', // non-standard
 'tabSize', 'textIndent', // non-standard
 'textRendering', 'textTransform', 'width', 'wordBreak'];
-var isIE = typeof document !== 'undefined' ? !!document.documentElement.currentStyle : false;
+var isIE = false;

 var getSizingData = function getSizingData(node) {
   var style = window.getComputedStyle(node);
