diff --git a/v4/core/core.cjs b/v4/core/core.cjs
index 937f35e921cc9e5db8b717699293fed65a5f0d33..120d5597b8475c12b8da96a2ea3185ce13112f9e 100644
--- a/v4/core/core.cjs
+++ b/v4/core/core.cjs
@@ -75,7 +75,8 @@ class $ZodEncodeError extends Error {
     }
 }
 exports.$ZodEncodeError = $ZodEncodeError;
-exports.globalConfig = {};
+// SIGNAL CHANGE
+exports.globalConfig = { jitless: true };
 function config(newConfig) {
     if (newConfig)
         Object.assign(exports.globalConfig, newConfig);
diff --git a/v4/core/core.js b/v4/core/core.js
index 2a32024b30237ea02204319642a4619c03ed82d1..613aa7b86d4df72ae5d3bceaed2b74ab787c660d 100644
--- a/v4/core/core.js
+++ b/v4/core/core.js
@@ -68,7 +68,8 @@ export class $ZodEncodeError extends Error {
         this.name = "ZodEncodeError";
     }
 }
-export const globalConfig = {};
+// SIGNAL CHANGE
+export const globalConfig = { jitless: true };
 export function config(newConfig) {
     if (newConfig)
         Object.assign(globalConfig, newConfig);
diff --git a/v4/core/registries.cjs b/v4/core/registries.cjs
index 3ab67ce0c138764cc5194e6f85881d9518cb0487..fc5994b29787bb8a1cb97db7ae2f65802f375a58 100644
--- a/v4/core/registries.cjs
+++ b/v4/core/registries.cjs
@@ -52,5 +52,6 @@ exports.$ZodRegistry = $ZodRegistry;
 function registry() {
     return new $ZodRegistry();
 }
-(_a = globalThis).__zod_globalRegistry ?? (_a.__zod_globalRegistry = registry());
-exports.globalRegistry = globalThis.__zod_globalRegistry;
+// SIGNAL CHANGE
+// (_a = globalThis).__zod_globalRegistry ?? (_a.__zod_globalRegistry = registry());
+exports.globalRegistry = registry();
diff --git a/v4/core/registries.js b/v4/core/registries.js
index 537b1f1b97c8c95c44485efbd143766dc9f29ead..3261b6812f3ed43d29f23c5e923ee955e6304c67 100644
--- a/v4/core/registries.js
+++ b/v4/core/registries.js
@@ -47,5 +47,6 @@ export class $ZodRegistry {
 export function registry() {
     return new $ZodRegistry();
 }
-(_a = globalThis).__zod_globalRegistry ?? (_a.__zod_globalRegistry = registry());
-export const globalRegistry = globalThis.__zod_globalRegistry;
+// SIGNAL CHANGE
+// (_a = globalThis).__zod_globalRegistry ?? (_a.__zod_globalRegistry = registry());
+export const globalRegistry = registry();
